membership.go

```diff
@@ -383,23 +383,29 @@ func multicastAnnounce(addr string) error {
 		return err
 	}
 
-	c, err := net.DialUDP("udp", nil, address)
-	if err != nil {
-		logError(err)
-		return err
-	}
+	for {
+		c, err := net.DialUDP("udp", nil, address)
+		if err != nil {
+			logError(err)
+			return err
+		}
 
-	// Compose and send the multicast announcement
-	msgBytes := encodeMulticastAnnounceBytes()
-	_, err = c.Write(msgBytes)
-	if err != nil {
-		logError(err)
-		return err
-	}
+		// Compose and send the multicast announcement
+		msgBytes := encodeMulticastAnnounceBytes()
+		_, err = c.Write(msgBytes)
+		if err != nil {
+			logError(err)
+			return err
+		}
 
-	logfTrace("Sent announcement multicast to %v\n", fullAddr)
+		logfTrace("Sent announcement multicast to %v\n", fullAddr)
 
-	return nil
+		if GetMulticastAnnounceInterval() > 0 {
```

Comment:
Just a nitpick: could you add a mention of the loops behavior in the function comments?